### PR TITLE
cleanup sysinfo package

### PR DESCRIPTION
**Pull request by cpuguy83**

### CODE REVIEW

**Comment by LK4D4** [comment]
@cpuguy83 Need rebase and CI failed.

**Comment by cpuguy83** [comment]
rebased, fixed CI issue.

File: pkg/sysinfo/sysinfo_linux.go [hunk 1]
```diff
@@ -13,82 +13,83 @@ import (
...
+}
 
-	// Check if AppArmor is supported.
-	if _, err := os.Stat("/sys/kernel/security/apparmor"); os.IsNotExist(err) {
-		sysInfo.AppArmor = false
-	} else {
-		sysInfo.AppArmor = true
-	}
+func cgroupEnabled(mountPoint, name string) bool {
+	_, err := ioutil.ReadFile(path.Join(mountPoint, name))
```
Comment (LK4D4) [inline pkg/sysinfo/sysinfo_linux.go]: isn't `ReadFile` overkill here. `os.Stat` should be enough.

Comment (cpuguy83) [inline pkg/sysinfo/sysinfo_linux.go]: Ah, yep.

File: pkg/sysinfo/sysinfo_linux.go [hunk 2]
```diff
@@ -13,82 +13,83 @@ import (
...
-	if _, err := cgroups.FindCgroupMountpoint("devices"); err != nil {
-		logrus.Fatalf("Error mounting devices cgroup: %v", err)
+func readProcBool(path string) bool {
+	val, err := ioutil.ReadFile(path)
+	if err != nil {
+		return false
 	}
 
-	return sysInfo
+	enabled, _ := strconv.Atoi(strings.TrimSpace(string(val)))
```
Comment (LK4D4) [inline pkg/sysinfo/sysinfo_linux.go]: how about `return string(val) == "1"`?

Comment (cpuguy83) [inline pkg/sysinfo/sysinfo_linux.go]: yeah buddy... I was focused on getting rid of `if{ ... } else { if {... } else {...}}` stuff, didn't even really start looking at these.

Will update.

**Comment by cpuguy83** [comment]
Updated

File: pkg/sysinfo/sysinfo_linux.go [hunk 3]
```diff
@@ -13,82 +11,81 @@ import (
 
 // New returns a new SysInfo, using the filesystem to detect which features the kernel supports.
 func New(quiet bool) *SysInfo {
+	// Check if Devices cgroup is mounted, it is hard requirement for container security.
+	if _, err := cgroups.FindCgroupMountpoint("devices"); err != nil {
```
Comment (LK4D4) [inline pkg/sysinfo/sysinfo_linux.go]: I don't know very well original code, but for me calling fatal in `pkg/` libraries isn't very cool. Also I'm not sure how and why `pkg/sysinfo` has hard requirement if `devices` cgroup.

Comment (cpuguy83) [inline pkg/sysinfo/sysinfo_linux.go]: Agreed, I can change this to return an error.
This originally comes from #11882

Comment (LK4D4) [inline pkg/sysinfo/sysinfo_linux.go]: Let's do this. Like maybe new field in `SysInfo`?

Comment (cpuguy83) [inline pkg/sysinfo/sysinfo_linux.go]: Done.

File: daemon/daemon.go [hunk 4]
```diff
@@ -743,6 +743,10 @@ func NewDaemon(config *Config, registryService *registry.Service) (daemon *Daemo
 	}
 
 	sysInfo := sysinfo.New(false)
+	if !sysInfo.CgroupDevicesEnabled {
+		return nil, fmt.Errorf("Error mounting cgroup devices")
```
Comment (LK4D4) [inline daemon/daemon.go]: Error is pretty weird. Should be like `Devices cgroup isn't mounted` or something like this. Also makes sense to write why it's bad.

Comment (cpuguy83) [inline daemon/daemon.go]: Done.

**Comment by LK4D4** [comment]
LGTM

**Comment by runcom** [comment]
LGTM

**Pull request merged**